libass/ass_directwrite.c

```diff
@@ -668,50 +670,49 @@ static void add_font(IDWriteFont *font, IDWriteFontFamily *fontFamily,
 }
 
 /*
- * Scan every system font on the current machine and add it
- * to the libass lookup. Stores the FontPrivate as private data
- * for later memory reading
+ * When a new font name is requested, called to load that font from Windows
  */
-static void scan_fonts(IDWriteFactory *factory,
-                       ASS_FontProvider *provider)
+static void match_fonts(void *priv, ASS_Library *lib,
+                        ASS_FontProvider *provider, char *name)
 {
-    HRESULT hr = S_OK;
-    IDWriteFontCollection *fontCollection = NULL;
+    ProviderPrivate *provider_priv = (ProviderPrivate *)priv;
     IDWriteFont *font = NULL;
-    hr = IDWriteFactory_GetSystemFontCollection(factory, &fontCollection, FALSE);
+    IDWriteFontFamily *fontFamily = NULL;
+    LOGFONTW lf;
+    HRESULT hr = S_OK;
 
-    if (FAILED(hr) || !fontCollection)
+    memset(&lf, 0, sizeof(LOGFONTW));
+    lf.lfEscapement = GM_COMPATIBLE;
+    lf.lfWeight = FW_DONTCARE;
+    lf.lfCharSet = DEFAULT_CHARSET;
+    lf.lfOutPrecision = OUT_TT_PRECIS;
+    lf.lfClipPrecision = CLIP_DEFAULT_PRECIS;
+    lf.lfQuality = DEFAULT_QUALITY;
```

Comment:
I suspect this probably doesn’t matter for our use case, but VSFilter sets this field to `ANTIALIASED_QUALITY`.

Reply:
IIRC this field affects the result of GDI rendered text. So it doesn't matter in our use case.

Reply:
Yeah, that’s what I think, too. But since I was editing this file anyway (to remove the `lfEscapement` just in case, and to remove the NUL termination just because), I’ve changed this to match VSFilter. At worst this should make no difference; at best this should help.